viewing_party/party.py

```diff
@@ -1,23 +1,137 @@
 # ------------- WAVE 1 --------------------
 
+
 def create_movie(title, genre, rating):
-    pass
+   movie ={}
+   if title and genre and rating :
+       movie["title"]=title
+       movie["genre"]=genre
+       movie["rating"]=rating
+       return movie
+   else:
+       return None
+   
+def  add_to_watched(user_data , movie):
+    user_data["watched"].append(movie)
+    return user_data
+    
+        
+def add_to_watchlist(user_data , movie):
+    user_data["watchlist"].append(movie)
+    return user_data
 
+def watch_movie(user_data , title):
+    for movie in user_data["watchlist"]:
+        
+      if movie["title"]==title:
+          user_data["watchlist"].remove(movie)
+          user_data["watched"].append(movie)
+          return user_data
+    else :
+        return user_data
+    
+  
+    
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating( user_data):
+    
+    while len(user_data["watched"])>0:
+        sum_rating=0
+        for movie in user_data["watched"]:
+            sum_rating+=movie["rating"]
+            
+        return sum_rating/len(user_data["watched"])
+    return 0.0
 
+    
+        
+def get_most_watched_genre(user_data):     
+    while len(user_data["watched"])>0:  
+        new_list=[]
+        for movie in user_data["watched"]:
+            new_list.append(movie["genre"])
+        return max(set(new_list), key=new_list.count)
+    return None
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    unique_watched=[]
+    unique_watched+=user_data["watched"]
+    #friends_watch+=user_data["friends"]
+    for list_1 in user_data["friends"] :
+        for movie in list_1["watched"]:

```

Comment:
let's come up with a more descriptive variable name than `list_1`. We want other coders to understand our code immediately. Well named variables can help with that.

Maybe call this `friend` since that is what the item is